server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesSourceParserHelper.java

```diff
@@ -81,11 +83,16 @@ public static void writeTo(CompositeValuesSourceBuilder<?> builder, StreamOutput
     }
 
     public static CompositeValuesSourceBuilder<?> fromXContent(XContentParser parser) throws IOException {
+        Matcher validAggMatcher = AggregatorFactories.VALID_AGG_NAME.matcher("");
         XContentParser.Token token = parser.currentToken();
         ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser);
         token = parser.nextToken();
         ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, parser);
         String name = parser.currentName();
+        if (validAggMatcher.reset(name).matches() == false) {
```

Comment:
Could you explain why you choose to instantiate the matcher with an empty patter and reset it later instead of just creating the matcher with the name?

Reply:
You're right. I'l fix that in my next commit.